packages/ui-kit/lib/components/video-player.tsx

```diff
@@ -0,0 +1,164 @@
+import MuxPlayer from '@mux/mux-player-react';
+import { useState, useEffect, useRef } from 'react';
+import { getDictionary, isLocalAware } from '@maany_shr/e-class-translations';
+
+export interface VideoPlayerProps extends isLocalAware {
+    videoId?: string;
+    thumbnailUrl?: string;
+    loadTimer?: number;
+    onErrorCallback?: (message: string, error: any) => void;
+}
+
+/**
+ * VideoPlayer Component
+ *
+ * A customizable video player that integrates with Mux for seamless video streaming.  
+ * It supports thumbnail previews, error handling, and localization.
+ *
+ * @component
+ * @param {VideoPlayerProps} props - The properties for the VideoPlayer component.
+ * @param {string} [props.videoId] - The unique identifier for the video (Mux playback ID).
+ * @param {string} [props.thumbnailUrl] - The URL of the thumbnail image displayed before playback.
+ * @param {string} props.locale - The current language/locale for text translations.
+ * @param {number} [props.loadTimer=3000] - The timeout in milliseconds before showing an error if video doesn't load.
+ * @param {(message: string, error: any) => void} props.onErrorCallback - A callback function triggered when an error occurs.
+ */
+export const VideoPlayer: React.FC<VideoPlayerProps> = ({
+    videoId,
+    thumbnailUrl,
+    locale,
+    loadTimer = 3000, // Default value of 3000ms (3 seconds)
+    onErrorCallback,
+}) => {
+    const dictionary = getDictionary(locale);
```

Comment:
I think it might be beneficial to utilize `useMemo` here to prevent recalculations.